Portability fixes for musl:

- _WORDIZE is declared in <bits/user.h> via <sys/user.h>.
- <xlocale.h> does not exist, check for it correctly.

	--xtraeme

--- configure.ac.orig	2015-09-24 11:36:06.107661526 +0200
+++ configure.ac	2015-09-24 11:36:40.420036775 +0200
@@ -407,7 +407,7 @@ AC_HEADER_STDC
 # POSIX
 AC_CHECK_HEADERS_ONCE([arpa/inet.h glob.h grp.h netdb.h netinet/in.h \
     netinet/in_systm.h netinet/tcp.h poll.h pwd.h sched.h \
-    sys/mman.h sys/select.h sys/socket.h sys/wait.h \
+    sys/mman.h sys/select.h sys/socket.h sys/wait.h sys/user.h xlocale.h \
     sys/uio.h syslog.h sys/dl.h dlfcn.h linux/sockios.h])
 AC_CHECK_HEADERS([netinet/ip.h], [], [],
                  [#include <sys/types.h>
--- src/pulsecore/sample-util.h.orig	2015-09-24 11:37:57.773677238 +0200
+++ src/pulsecore/sample-util.h	2015-09-24 11:38:46.668208677 +0200
@@ -24,6 +24,14 @@
 #include <inttypes.h>
 #include <limits.h>
 
+#ifdef HAVE_CONFIG_H
+#include <config.h>
+#endif
+
+#ifdef HAVE_SYS_USER_H
+#include <sys/user.h>	/* required by _WORDIZE with musl */
+#endif
+
 #include <pulse/gccmacro.h>
 #include <pulse/sample.h>
 #include <pulse/volume.h>
--- src/pulsecore/core-util.c.orig	2015-09-24 11:45:57.919919956 +0200
+++ src/pulsecore/core-util.c	2015-09-24 11:46:23.343196284 +0200
@@ -54,7 +54,9 @@
 
 #ifdef HAVE_STRTOD_L
 #include <locale.h>
-#include <xlocale.h>
+# ifdef HAVE_XLOCALE_H
+#  include <xlocale.h>
+# endif
 #endif
 
 #ifdef HAVE_SCHED_H
